csharp/src/Apache.Arrow/C/CArrowArrayExporter.cs

```diff
@@ -203,6 +207,10 @@ private unsafe static void ReleaseArray(CArrowArray* cArray)
         private unsafe static void Dispose(void** ptr)
         {
             GCHandle gch = GCHandle.FromIntPtr((IntPtr)(*ptr));
+            if (!gch.IsAllocated)
```

Comment:
Add a comment explaining when this might occur? (perhaps if an exception occurs while exporting the array?)